pandas/core/dtypes/cast.py

```diff
@@ -357,15 +357,11 @@ def trans(x):
             # if we don't have any elements, just astype it
             return trans(result).astype(dtype)
 
-        # do a test on the first element, if it fails then we are done
-        r = result.ravel()
-        arr = np.array([r[0]])
-
-        if isna(arr).any():
-            # if we have any nulls, then we are done
-            return result
-
-        elif not isinstance(r[0], (np.integer, np.floating, int, float, bool)):
+        if isinstance(result, np.ndarray):
```

Comment:
In what case are we dealing with something that isn't a NumPy array?

Reply:
`pd.interval_range` seems to be the common organic reason.  `tests/dtypes/cast/test_downcast` file triggers the case by explicitly creating `ndarray` objects.

Reply:
It looks like we get here with Series, which isn't part of the type-annotations. That should probably be fixed, but outside the scope here.